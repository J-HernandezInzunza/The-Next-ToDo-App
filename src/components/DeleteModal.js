import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';

const AlertDialog = ({ showModal, selectedItem, toggleModal, onTodoDelete }) => {
  return (
    <>
      <Dialog open={showModal} onClose={() => toggleModal()}>
        <DialogTitle>Do you wish to delete this Item?</DialogTitle>
        <DialogContent>
          <DialogContentText>
            {`Delete the todo with text: "${selectedItem ? selectedItem.text : ''}"`}
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={() => toggleModal()} color="primary">
            Keep It
          </Button>
          <Button onClick={() => onTodoDelete(selectedItem)} color="secondary" autoFocus>
            Trash It
          </Button>
        </DialogActions>
      </Dialog>
    </>
  );
};

export default AlertDialog;
