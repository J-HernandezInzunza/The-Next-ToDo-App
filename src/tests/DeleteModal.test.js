import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import DeleteModal from '../components/DeleteModal';

afterEach(cleanup);

describe('DeleteModal Tests', () => {
  it('Renders correctly', () => {
    const { getByTestId, asFragment } = render(<DeleteModal showModal={true} />);
    expect(asFragment()).toMatchSnapshot();
    expect(getByTestId('dialog-title')).toHaveTextContent(`delete this Item?`);
  });

  it('Inserts text into dialog content', () => {
    const staticDeleteText = 'Delete the todo with text:';
    const dynamicDeleteText = 'Hello Moto';
    const selectedItem = { text: dynamicDeleteText };
    const { getByTestId } = render(<DeleteModal showModal={true} selectedItem={selectedItem} />);

    expect(getByTestId('dialog-text')).toHaveTextContent(
      `${staticDeleteText} "${dynamicDeleteText}"`
    );
  });

  it('Calls functions when "Keep It" and "Trash It" buttons are pressed', () => {
    const toggleModal = jest.fn();
    const onTodoDelete = jest.fn();
    const { getByText } = render(
      <DeleteModal showModal={true} toggleModal={toggleModal} onTodoDelete={onTodoDelete} />
    );

    fireEvent.click(getByText('Keep It'));
    expect(toggleModal).toHaveBeenCalledTimes(1);

    fireEvent.click(getByText('Trash It'));
    expect(onTodoDelete).toHaveBeenCalledTimes(1);
  });
});
