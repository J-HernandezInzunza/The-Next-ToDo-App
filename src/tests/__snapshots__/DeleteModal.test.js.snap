// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteModal Tests Renders correctly 1`] = `<DocumentFragment />`;
